Added an example how to deploy resources only for specific targets (#96)
databricks/bundle-examples: knowledge_base/target_includes/README.md

# Target Includes Example

This example demonstrates the concept of using `target_includes` (or similar include mechanisms) in Databricks Asset Bundles to organize job configurations across different environments without duplication.

> ## Overview
> 
> This example implements the use case described in [GitHub Issue #2878](https://github.com/databricks/cli/issues/2878), which requests the ability to include specific resource files based on target configurations.

Reviewer: ```suggestion
It implements the use case described in [GitHub Issue #2878](https://github.com/databricks/cli/issues/2878), which requests the ability to include specific resource files based on target configurations.
```
(don't need to have an overview heading because all of this intro content is overview)

## Directory Structure

```
target_includes/
├── databricks.yml          # Main bundle configuration with 3 targets
├── resources/
│   ├── set1/              # Jobs for dev and staging environments
│   │   ├── job_1.yml
│   │   └── job_2.yml
│   └── set2/              # Jobs for staging and prod environments
│       ├── job_1.yml
│       └── job_2.yml
└── README.md
```

## Target Configuration

The bundle defines three targets:

> 1. **dev**: Includes only `resources/set1/*.yml`
>    - Contains: set1-job-1, set1-job-2
>    - Environment: development
> 
> 2. **staging**: Includes both `resources/set1/*.yml` and `resources/set2/*.yml`
>    - Contains: set1-job-1, set1-job-2, set2-job-1, set2-job-2
>    - Environment: staging
> 
> 3. **prod**: Includes only `resources/set2/*.yml`
>    - Contains: set2-job-1, set2-job-2
>    - Environment: production

Reviewer: ```suggestion
- **dev**: Includes only `resources/set1/*.yml`
   - Contains: set1-job-1, set1-job-2
   - Environment: development

- **staging**: Includes both `resources/set1/*.yml` and `resources/set2/*.yml`
   - Contains: set1-job-1, set1-job-2, set2-job-1, set2-job-2
   - Environment: staging

- **prod**: Includes only `resources/set2/*.yml`
   - Contains: set2-job-1, set2-job-2
   - Environment: production
```
nit, they aren't steps

## Usage

```bash
# Summary of the bundle resources for dev
databricks bundle summary -p u2m -t dev

> Name: target-includes-example

Author: Fixed here: https://github.com/databricks/bundle-examples/pull/96/commits/1570ba493896c528425b4c815657a99c8de4a002

Target: dev
Workspace:
  User: ***
  Path: ***
Resources:
  Jobs:
    set1-job-1:
      Name: Set1 Job 1 - foo
      URL:  (not deployed)
    set1-job-2:
      Name: Set1 Job 2 - foo
      URL:  (not deployed)

# Summary of the bundle resources for staging
databricks bundle summary -p u2m -t staging

Name: target-includes-example
Target: staging
Workspace:
  User: ***
  Path: ***
Resources:
  Jobs:
    set1-job-1:
      Name: Set1 Job 1 - bar
      URL:  (not deployed)
    set1-job-2:
      Name: Set1 Job 2 - bar
      URL:  (not deployed)
    set2-job-1:
      Name: Set2 Job 1 - bar
      URL:  (not deployed)
    set2-job-2:
      Name: Set2 Job 2 - bar
      URL:  (not deployed)

# Summary of the bundle resources for prod
databricks bundle summary -p u2m -t prod   

Name: target-includes-example
Target: prod
Workspace:
  User: ***
  Path: ***
Resources:
  Jobs:
    set2-job-1:
      Name: Set2 Job 1 - baz
      URL:  (not deployed)
    set2-job-2:
      Name: Set2 Job 2 - baz
      URL:  (not deployed)
```

> ## Notes
> 
> There are some key aspects in this implementation
> 1. In `databricks.yml` file we include (see `include` section) all configuration files for all targets. This does not impact which resources will be deployed for which target.
> 2. For each job in corresponding configuration file like `resources/set1/job_1.yml` we define in which targets this job should be deployed. We use YAML anchors to avoid duplications between targets.

Reviewer: ```suggestion
## Notes

There are some important aspects of this implementation:
- The `databricks.yml` file includes all configuration files for all targets (see `include` section). This does not impact which resources will be deployed in each target.
- For each job in a corresponding configuration file, such as `resources/set1/job_1.yml`, targets are defined where the job should be deployed. YAML anchors are used to avoid duplications between targets.
```
Edits, and i try to avoid the word "key" if it means anything other than code element (YAML, auth, etc.).

I would move this section to before Usage too.
